keras/datasets/imdb.py

```diff
@@ -94,38 +84,17 @@ def load_data(path='imdb.npz', num_words=None, skip_top=0,
     # reserve 'index_from' (=3 by default) characters:
     # 0 (padding), 1 (start), 2 (OOV)
     if oov_char is not None:
-        xs = [[oov_char if (w >= num_words or w < skip_top) else w for w in x] for x in xs]
+        xs = [[w if (skip_top <= w < num_words) else oov_char for w in x] for x in xs]
     else:
-        new_xs = []
-        for x in xs:
-            nx = []
-            for w in x:
-                if skip_top <= w < num_words:
-                    nx.append(w)
-            new_xs.append(nx)
-        xs = new_xs
+        xs = [[w for w in x if (skip_top <= w < num_words)] for x in xs]
 
-    x_train = np.array(xs[:len(x_train)])
-    y_train = np.array(labels[:len(x_train)])
-
-    x_test = np.array(xs[len(x_train):])
-    y_test = np.array(labels[len(x_train):])
+    idx = len(x_train)
+    x_train, y_train = np.array(xs[:idx]), np.array(labels[:idx])
+    x_test, y_test = np.array(xs[idx:]), np.array(labels[idx:])
 
     return (x_train, y_train), (x_test, y_test)
 
 
 def get_word_index(path='imdb_word_index.json'):
-    """Retrieves the dictionary mapping word indices back to words.
-
-    # Arguments
-        path: where to cache the data (relative to `~/.keras/dataset`).
-
-    # Returns
-        The word index dictionary.
-    """
```

Comment:
You should not remove the docstring, it is potentially useful for generating docs. Also I don't think you need to abstract this function, it's just 3 less lines